docs: How to Use Prisma Client Extensions with DB Client and Logging (#9283)
redwoodjs/redwood: docs/docs/logger.md

})
```

### Prisma Client Extensions

> Because [$on and $use are not available in extended clients](https://www.prisma.io/docs/concepts/components/prisma-client/client-extensions#usage-of-on-and-use-with-extended-clients), if you would like to continue using these client-level methods with an extended client, you will need to hook them up before [extending the client](https://www.prisma.io/docs/concepts/components/prisma-client/client-extensions#extended-clients.

Reviewer: this isn't rendering correctly because of a missing `)`:

```suggestion
Because [$on and $use are not available in extended clients](https://www.prisma.io/docs/concepts/components/prisma-client/client-extensions#usage-of-on-and-use-with-extended-clients), if you would like to continue using these client-level methods with an extended client, you will need to hook them up before [extending the client](https://www.prisma.io/docs/concepts/components/prisma-client/client-extensions#extended-clients).
```

> This requires a slight change to how your Prisma client is created, logging added, and then the `db` is exported.

Reviewer: just read a little awkwardly and tried to improve it:

```suggestion
This requires a slight change to how your Prisma client is created, configured for logging, and exported as `db`:
```

```ts
import { PrismaClient } from '@prisma/client'

import {
  LogLevel,
  emitLogLevels,
  handlePrismaLogging,
} from '@redwoodjs/api/logger'

import { logger } from './logger'

const logLevels = process.env.DB_LOGGING
  ? (['info', 'warn', 'error', 'query'] as LogLevel[])
  : (['info', 'warn', 'error'] as LogLevel[])

/*
 * Instance of the Prisma Client
 */
const client = new PrismaClient({
  log: emitLogLevels(logLevels),
})

handlePrismaLogging({
  db: client,
  logger,
  logLevels,
})

export const db = client.$extends({
  query: {
    $allModels: {
      async $allOperations({ operation, model, args, query }) {
        const start = performance.now()
        const result = await query(args)
        const end = performance.now()
        const time = end - start
        console.log(JSON.stringify({ operation, model, args, time }, null, 2))
        return result
      },
    },
  },
})
```

> Now the Prisma client will log the designated emited log levels as well as invoke the client extension:

Reviewer: ```suggestion
Now the Prisma client will log the specified emitted log levels as well as invoke the client extension:
```
